pylabrobot/liquid_handling/liquid_handler.py

```diff
@@ -1383,6 +1399,21 @@ def save(self, path: str):
     with open(path, "w", encoding="utf-8") as f:
       json.dump(self.serialize(), f, indent=2)
 
+  def register_callback(self, method_name: str, callback: OperationCallback):
+    """Registers a callback for a specific method."""
```

Comment:
I like that you decided to use strings to identify callbacks, simplifies things a lot.

I think this method should check that the method_name is allowed (`method_name in {'aspirate', 'dispense', etc.}`.

Reply:
Let me know if you like the implementation, and if the list of allowed method names is correct.

Reply:
The ALLOWED_CALLBACKS constant is great!